cpp/src/io/parquet/experimental/hybrid_scan_helpers.cpp

```diff
@@ -137,4 +149,110 @@ void aggregate_reader_metadata::setup_page_index(cudf::host_span<uint8_t const>
   }
 }
 
+std::tuple<std::vector<input_column_info>,
+           std::vector<inline_column_buffer>,
+           std::vector<cudf::size_type>>
+aggregate_reader_metadata::select_payload_columns(
```

Comment:
do we (or can we) have tests for different code paths in this function? One thing that comes to mind are the different combinations of `payload_column_names`/`filter_column_names` with and without value, as well as having nested columns in `filter_column_names`.

Reply:
This one is a bit tricky to test now since, as of this PR, we only have row group pruning APIs implemented which call `hybrid_scan_impl::select_columns` with `read_mode::FILTER_COLUMNS` and hence not calling this `select_payload_columns()` function at all. We can test this once we add `materialize_payload_columns` API implementation.